ee/hogai/graph/filter_options/toolkit.py

```diff
@@ -0,0 +1,257 @@
+
+        result = {"properties": properties_by_type}
+        return yaml.dump(result, default_flow_style=False, sort_keys=True)
+
+    def _enrich_props_with_descriptions(self, entity: str, props: Iterable[tuple[str, str | None]]):
+        enriched_props = []
+        mapping = {
+            "session": CORE_FILTER_DEFINITIONS_BY_GROUP["session_properties"],
+            "person": CORE_FILTER_DEFINITIONS_BY_GROUP["person_properties"],
+            "event": CORE_FILTER_DEFINITIONS_BY_GROUP["event_properties"],
+        }
+        for prop_name, prop_type in props:
+            description = None
+            if entity_definition := mapping.get(entity, {}).get(prop_name):
+                if entity_definition.get("system") or entity_definition.get("ignored_in_assistant"):
+                    continue
+                description = entity_definition.get("description_llm") or entity_definition.get("description")
+            enriched_props.append((prop_name, prop_type, description))
+        return enriched_props
+
+    def _format_property_values(
+        self, sample_values: list, sample_count: Optional[int] = 0, format_as_string: bool = False
+    ) -> str:
+        if len(sample_values) == 0 or sample_count == 0:
+            return f"The property does not have any values in the taxonomy. Use the value that the user has provided in the query."
+
+        # Add quotes to the String type, so the LLM can easily infer a type.
+        # Strings like "true" or "10" are interpreted as booleans or numbers without quotes, so the schema generation fails.
+        # Remove the floating point the value is an integer.
+        formatted_sample_values: list[str] = []
+        for value in sample_values:
+            if format_as_string:
+                formatted_sample_values.append(f'"{value}"')
+            elif isinstance(value, float) and value.is_integer():
+                formatted_sample_values.append(str(int(value)))
+            else:
+                formatted_sample_values.append(str(value))
+        prop_values = ", ".join(formatted_sample_values)
+
+        # If there wasn't an exact match with the user's search, we provide a hint that LLM can use an arbitrary value.
+        if sample_count is None:
+            return f"{prop_values} and many more distinct values."
+        elif sample_count > len(sample_values):
+            diff = sample_count - len(sample_values)
+            return f"{prop_values} and {diff} more distinct value{'' if diff == 1 else 's'}."
+
+        return prop_values
+
+    def retrieve_entity_property_values(self, entity: str, property_name: str) -> str:
+        """
+        Retrieve property values for an entity and property name.
+        """
+        MAX_PROP_VALUES = 50
+        if entity not in self._entity_names:
+            return f"The entity {entity} does not exist in the taxonomy. Try one of these entities: {', '.join(self._entity_names)}."
```

Comment:
These tools seem highly duplicated from TaxonomyAgentToolkit. Can we unify these? The definitions like `class retrieve_entity_properties` could be imported outright, while perhaps the methods like this one could be a mixin? Or even TaxonomyAgentToolkit could inherit from FilterOptionsToolkit, or vice versa